### PR TITLE
Feat/findsqlpp11

### LINKED ISSUE #44: find_package and sqlpp11
Hi,

I was wondering why this library was not calling `find_package(sqlpp11)` as it's doing right now with `HinnantDate`, `Sqlite3`, and `SQLCipher`. That would make more sense IMO, and remove the duplicated `FindHinnantDate.cmake` file, which is already present in `sqlpp11/cmake`.

Also, I'm using Conan and I had to heavily patch the `CMakeLists.txt` to use my previously built dependencies (sqlcipher etc...). Having a `find_package` for each dependency would reduce the number of patches.

What do you think about it?

### PR DESCRIPTION
**Pull request by theodelrieu**
Hi!

Sorry for the long wait, this PR should close #44.

* use `find_package` to link against sqlpp11.
* export cmake targets and files for consumers.

Based on https://github.com/rbock/sqlpp11/pull/225

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,19 +24,17 @@
 
 cmake_minimum_required(VERSION 3.2)
 include(CheckCXXSymbolExists)
-project (sqlpp11-connector-sqlite3)
+project (sqlpp11-connector-sqlite3 VERSION 0.28 LANGUAGES CXX)
```
Comment (rbock) [inline CMakeLists.txt]: Can you think of a way to avoid putting the version number here?

Versions are tagged in git. I would not want to have to keep the CMakeLists in sync (I will forget to do that for sure).

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@ add_library(sqlpp11-connector-sqlite3
 		detail/connection_handle.cpp
       )
 
-if (SQLPP_DYNAMIC_LOADING)
+if (DYNAMIC_LOADING)
```
Comment (rbock) [inline src/CMakeLists.txt]: Why did you remove the prefix?

File: CMakeLists.txt (hunk 1 above)
Comment (theodelrieu) [reply CMakeLists.txt]: At first I tried to mimick what was in sqlpp11 CMakeLists.txt. But the version was 0.1.

I removed the version file in the end.

File: src/CMakeLists.txt (hunk 2 above)
Comment (theodelrieu) [reply src/CMakeLists.txt]: There was a mismatch between the `SQLPP_DYNAMIC_LOADING` compile flag, and the cmake option. I forgot to put it back.

Fixed in latest commit.

**Comment by theodelrieu** [comment]
I thought I could remove the Find modules in `sqlpp11-connector-sqlite3`, but it wasn't the case.

TIL CMake needs them when libraries are not installed with cmake.

**Comment by rbock** [comment]
The sqlite3 connector got merged into sqlpp11, making this obsolete.
